### PR TITLE
Removing noise model check in Simulator initialization

### PR DESCRIPTION
**Pull request by asmuzsoy**
Addresses issue #4170 by removing NOISE_MODEL_LIKE check in sparse_simulator.py and updates tests accordingly.

### CODE REVIEW

**Comment by asmuzsoy** [comment]
@googlebot I signed it!

File: cirq-core/cirq/sim/sparse_simulator_test.py [hunk 1]
```diff
@@ -1270,6 +1301,20 @@ def test_nondeterministic_mixture_noise():
...
+        depol_noise=True,
+        readout_decay_noise=True,
+        readout_error_noise=True,
+    )
+    simulator = cirq.Simulator(noise=noise_model)
+
+    result1 = simulator.run(circuit, repetitions=50)
+    result2 = simulator.run(circuit, repetitions=50)
+
+    assert result1 != result2
```
Comment (viathor) [inline cirq-core/cirq/sim/sparse_simulator_test.py:1320]: I suppose for this PR we just want to verify that the simulator accepts the noise model, as opposed to raising to an exception. However, we can make the test more useful by checking that the measurement falls into an interval (which ideally would be verified by a calculation independent of the code under test).

**Review by viathor — COMMENTED** [review]
Please merge the master branch back into this one.

File: cirq-core/cirq/sim/sparse_simulator_test.py [hunk 2]
```diff
@@ -1270,6 +1270,14 @@ def test_nondeterministic_mixture_noise():
...
+def test_noise_model():
+    q = cirq.LineQubit(0)
+    circuit = cirq.Circuit(cirq.I(q), cirq.measure(q))
+
+    noise_model = cirq.NoiseModel.from_noise_model_like(cirq.depolarize(p=0.01))
+
+    simulator = cirq.Simulator(noise=noise_model)
+
+    result = simulator.run(circuit, repetitions=100)
+
```
Comment (viathor) [inline cirq-core/cirq/sim/sparse_simulator_test.py:1282]: nit: this is a very short (six lines) test, so we don't need so much vertical whitespace :-) let's remove the empty lines

File: cirq-core/cirq/sim/sparse_simulator_test.py [hunk 3]
```diff
@@ -1270,6 +1270,14 @@ def test_nondeterministic_mixture_noise():
     assert result1 != result2
 
 
-def test_unsupported_noise_fails():
-    with pytest.raises(ValueError, match='noise'):
-        cirq.Simulator(noise=cirq.amplitude_damp(0.5))
+def test_noise_model():
+    q = cirq.LineQubit(0)
+    circuit = cirq.Circuit(cirq.I(q), cirq.measure(q))
```
Comment (viathor) [inline cirq-core/cirq/sim/sparse_simulator_test.py:1275]: There is a class of bugs that make `sum(result.measurements['0'])[0]` equal to zero exactly (e.g. one such bug is the simulator accepting but ignoring the noise). These bugs would pass undetected by the test.

Suggestion: Replace identity with `cirq.H(q)` and change the assert to something like

```python
assert 40 <= sum(result.measurements['0'])[0] < 60
```

Feel free to use tighter bounds, I just made those up. They should be gapped away from zero on the left and away from 100 on the right, though.

Comment (asmuzsoy) [reply cirq-core/cirq/sim/sparse_simulator_test.py]: thanks for the suggestion, I just updated it!

**Review by viathor — APPROVED** [review]

**Pull request merged**